components/script/script_thread.rs

```diff
@@ -2651,7 +2651,13 @@ impl ScriptThread {
     }
 
     fn perform_a_microtask_checkpoint(&self) {
-        self.microtask_queue.checkpoint(|id| self.documents.borrow().find_global(id))
+        self.microtask_queue.checkpoint(
+            |id| self.documents.borrow().find_global(id),
+            self.documents.borrow()
+                            .iter()
+                            .filter_map(|(id, _doc)| self.documents.borrow().find_global(id))
```

Comment:
What if we use map and return `document.global()` instead?

Reply:
Oh! Sounds good idea! I will try it! Thanks!